src/services/BavService.ts

```diff
@@ -0,0 +1,180 @@
+				throw new AppError(HttpCodesEnum.UNAUTHORIZED, `Session with session id: ${sessionId} has expired`);
+			}
+			return session.Item as ISessionItem;
+		}
+	}
+
+	async sendToTXMA(QueueUrl: string, event: TxmaEvent): Promise<void> {
+		try {
+			const messageBody = JSON.stringify(event);
+			const params = {
+				MessageBody: messageBody,
+				QueueUrl,
+			};
+
+			this.logger.info({ message: "Sending message to TxMA", eventName: event.event_name });
+
+			await sqsClient.send(new SendMessageCommand(params));
+			this.logger.info("Sent message to TxMA");
+		} catch (error) {
+			this.logger.error({
+				message: `Error when sending event ${event.event_name} to TXMA Queue`,
+				error,
+				messageCode: MessageCodes.FAILED_TO_WRITE_TXMA,
+			});
+		}
+	}
+
+	private mapBirthDate(birthDate: PersonIdentityDateOfBirth[]): PersonIdentityDateOfBirth[] {
+		return birthDate?.map((bd) => ({ value: bd.value }));
+	}
+
+	private mapNames(name: PersonIdentityName[]): PersonIdentityName[] {
+		return name?.map((index) => ({
+			nameParts: index?.nameParts?.map((namePart) => ({
+				type: namePart.type,
+				value: namePart.value,
+			})),
+		}));
+	}
+
+	private createPersonIdentityItem(
+		sharedClaims: SharedClaimsPersonIdentity,
+		sessionId: string,
+		authSessionTtlInSecs: string,
+	): PersonIdentityItem {
+
+		return {
+			sessionId,
+			birthDate: this.mapBirthDate(sharedClaims.birthDate),
+			name: this.mapNames(sharedClaims.name),
+  		expiryDate: absoluteTimeNow() + +authSessionTtlInSecs,
+			createdDate: absoluteTimeNow(),
+		};
+	}
+
+	async savePersonIdentity({
+		sharedClaims,
+		sessionId,
+		tableName,
+		authSessionTtlInSecs,
+	}: {
+		sharedClaims: SharedClaimsPersonIdentity;
+		sessionId: string;
+		tableName: string;
+		authSessionTtlInSecs: string;
+	}): Promise<void> {
+		const personIdentityItem = this.createPersonIdentityItem(
+			sharedClaims,
+			sessionId,
+			authSessionTtlInSecs,
+		);
+
+		const putSessionCommand = new PutCommand({
+			TableName: tableName,
+			Item: personIdentityItem,
+		});
+
+		this.logger.info({ message: "Saving personal identity data in DynamoDB" });
+
+		try {
+			await this.dynamo.send(putSessionCommand);
+			this.logger.info("Successfully created session in dynamodb");
+			return putSessionCommand?.input?.Item?.sessionId;
+
+		} catch (error) {
+			this.logger.error({ message: "Failed to save personal identity information", error });
+			throw new AppError(HttpCodesEnum.SERVER_ERROR, "Failed to save personal identity information" );
+		}
+	}
+
+	async createAuthSession(session: ISessionItem): Promise<void> {
+		const putSessionCommand = new PutCommand({
+			TableName: this.tableName,
+			Item: session,
+		});
+
+		this.logger.info({ message: "Saving session data in DynamoDB" });
+
+		try {
+			await this.dynamo.send(putSessionCommand);
+			this.logger.info("Successfully created session in dynamodb");
```

Comment:
Successfully saved auth data in DynamoDB?

Reply:
Yes agree - done